File: 725-split-linked-list-in-parts/split-linked-list-in-parts.py
class Solution:
    def splitListToParts(
        self, head: Optional[ListNode], k: int
    ) -> List[Optional[ListNode]]:
        curr, L = head, 0
        while curr:
            L += 1
            curr = curr.next

        minContain = L // k
        remainderNodes = L % k

        result = [None] * k
        curr = head
        prev = None

        for i in range(k):
            result[i] = curr
            for count in range(1, minContain + (1 if remainderNodes > 0 else 0) + 1):
                prev = curr
                if curr:
                    curr = curr.next
            if prev:
                prev.next = None
            remainderNodes -= 1
        return result

File: 725-split-linked-list-in-parts/split-linked-list-in-parts.js
var splitListToParts = function (head, k) {
    let curr = head
    let L = 0
    while (curr) {
        L++;
        curr = curr.next
    }
    const minContain = Math.floor(L / k)
    let remainder = L % k

    const result = new Array(k).fill(null)
    curr = head
    let prev = null

    for (let i = 0; curr !== null && i < k; i++) {
        result[i] = curr
        for (let count = 1; count <= minContain + (remainder > 0 ? 1 : 0); count++) {
            prev = curr
            curr = curr.next
        }
        if (prev !== null) {
            prev.next = null
        }
        remainder--
    }
    return result
};